internal/provider/resource.go

```diff
@@ -5,47 +5,64 @@ import (
 	"fmt"
 	"math/rand"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func nullResource() *schema.Resource {
-	return &schema.Resource{
-		Description: `The ` + "`null_resource`" + ` resource implements the standard resource lifecycle but takes no further action.
+var (
+	_ resource.Resource = (*nullResource)(nil)
+)
 
-The ` + "`triggers`" + ` argument allows specifying an arbitrary set of values that, when changed, will cause the resource to be replaced.`,
+func NewNullResource() resource.Resource {
+	return &nullResource{}
+}
+
+type nullResource struct{}
+
+func (n nullResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
```

Comment:
Nit: For consistency, should the `nullResource` functions be on a pointer receiver as `NewNullResource` returns a pointer?